tests/model/test_item.py

```diff
@@ -136,3 +138,105 @@ def test_resolve_item_parameters():
+    for label in labels:
+        add_to_initial(label, parameters, initial_parameters)
+
+
+def _test_add_to_initial(initial_parameters):
+    def check_parameter_order(parameters, expected_order):
+        actual_order = list(parameters._parameters.keys())
+        assert (
+            actual_order == expected_order
+        ), f"Parameters not in expected order. Got: {actual_order}, Expected: {expected_order}"
+
+    def test_ordered_addition():
+        parameters = Parameters({})
+        _add_parameters_to_initial(
+            ["b.1", "b.2", "rates.k1", "rates.k2"], parameters, initial_parameters
+        )
+        check_parameter_order(parameters, ["b.1", "b.2", "rates.k1sum", "rates.k1", "rates.k2"])
+
+    def test_unordered_addition():
+        parameters = Parameters({})
+        _add_parameters_to_initial(
+            ["b.2", "b.1", "rates.k1", "rates.k2"], parameters, initial_parameters
+        )
+        check_parameter_order(parameters, ["b.1", "b.2", "rates.k1sum", "rates.k1", "rates.k2"])
+
+    def test_partial_addition():
+        parameters = Parameters({})
+        _add_parameters_to_initial(["rates.k1", "rates.k2"], parameters, initial_parameters)
+        check_parameter_order(parameters, ["b.1", "rates.k1sum", "rates.k1", "b.2", "rates.k2"])
+
+    def test_randomized_addition():
+        parameter_labels = ["b.1", "b.2", "rates.k1", "rates.k2", "rates.k3"]
+        random.shuffle(parameter_labels)
+        parameters = Parameters({})
+        for label in parameter_labels:
+            add_to_initial(label, parameters, initial_parameters)
+
+        keys_list = list(parameters._parameters.keys())
+
+        assert keys_list.index("rates.k2") > keys_list.index(
+            "b.2"
+        ), "rates.k2 should come after b.2"
+        assert keys_list.index("b.2") > keys_list.index("b.1"), "b.2 should come after b.1"
+        assert keys_list.index("rates.k1") > keys_list.index(
+            "rates.k1sum"
+        ), "rates.k1 should come after rates.k1sum"
+        assert keys_list.index("rates.k2") > keys_list.index(
+            "rates.k1sum"
+        ), "rates.k2 should come after rates.k1sum"
+
+        return parameters
+
+    def check_expressions(parameters):
+        assert (
+            parameters.get("b.2").expression == "1.0 - $b.1"
+        ), f"Incorrect expression for b.2. Got: {parameters.get('b.2').expression}"
+        assert (
+            parameters.get("rates.k1").expression == "$b.1 * $rates.k1sum"
+        ), f"Incorrect expression for rates.k1. Got: {parameters.get('rates.k1').expression}"
+        assert (
+            parameters.get("rates.k2").expression == "$b.2 * $rates.k1sum"
+        ), f"Incorrect expression for rates.k2. Got: {parameters.get('rates.k2').expression}"
+
+    test_ordered_addition()
+    test_unordered_addition()
+    test_partial_addition()
+    parameters = test_randomized_addition()
+    check_expressions(parameters)
+
+
+def test_add_to_initial_b_first():
```

Comment:
Can you please add docstrings to the tests that explain the difference (in order vs. out of order definition)

Reply:
Addressed in [fe69696](https://github.com/glotaran/pyglotaran/pull/1500/commits/fe69696ed3be26218c0d80c756c62dbcc3045f81)